minimal.js

```diff
@@ -208,34 +221,48 @@ module.exports = function (dirname, keys, opts) {
     else flush.push(cb)
   }
 
+  db.addBoxer = function (boxer) {
+    boxers.push(boxer)
+  }
+
   db.addUnboxer = function (unboxer) {
     unboxers.push(unboxer)
   }
 
-  db.unbox = function (data, key) {
-    return unbox(data, unboxers, key)
-  }
-  db.box = function (opts, state) {
-    var content = opts.content
-    var recps = opts.content.recps
-    if (recps) {
-      const isNonEmptyArrayOfFeeds = isArray(recps) && recps.every(isFeed) && recps.length > 0
-      if (isFeed(recps) || isNonEmptyArrayOfFeeds) {
-        recps = opts.content.recps = [].concat(recps) // force to array
-        return box(opts.content, recps)
-      } else {
-        throw new Error(
-          'private message recipients must be valid feeds, '+
-          'was:' + JSON.stringify(recps)
-        )
-      }
+  /* TODO extract to ssb-private */
+  var box1 = {
+    box: function (content, recps) {
+      if (!recps.every(isFeed)) return null
+
+      return ssbKeys.box(content, recps)
+    },
+    unbox: {
+      key: function (ciphertext) { return ssbKeys.unboxKey(ciphertext, keys) },
+      value: function (ciphertext, key) { return ssbKeys.unboxBody(ciphertext, key) }
     }
-    return opts.content
   }
+  db.addBoxer(box1.box)
+  db.addUnboxer(box1.unbox)
+  // ////////////////////////////////
+
+  // db.box = function (value, state) {
+  db.box = function (value) {
+    var recps = value.content.recps
+    if (!recps) return value.content
+
+    if (typeof recps === 'string') recps = value.content.recps = [recps]
+    if (!Array.isArray(recps)) throw new Error('private message field "recps" expects an Array of recipients')
```

Comment:
Maybe redundant, but my intuition is that this change would throw errors when Patchbay makes private messages. If that's true, it would be important to ensure that this is published as a breaking change. FWIW, I'm a fan of the simplicity.

Reply:
I don't see how this make a breaking change. See above comment

Reply:
Yup, my bad there.